includes/formFactory/ManageWikiFormFactory.php

```diff
@@ -380,60 +412,29 @@ protected function submitForm(
 			$mwStore = 'mw_namespaces';
 			$build = [];
 
+			$nsID = [
+				'namespace' => (int)$special,
+				'namespacetalk' => (int)$special + 1
+			];
+
 			$existingNamespace = $dbw->selectRow(
 				'mw_namespaces',
-				'ns_namespace_id',
+				'ns_namespace_name',
 				[
 					'ns_dbname' => $wgDBname,
-					'ns_namespace_name' => str_replace( ' ', '_', $formData['namespace-namespace'] )
+					'ns_namespace_id' => $nsID['namespace']
 				],
 				__METHOD__
 			);
 
-			if ( $existingNamespace ) {
-				$nsID = [
-					'namespace' => (int)$existingNamespace->ns_namespace_id,
-					'namespacetalk' => (int)$existingNamespace->ns_namespace_id + 1
-				];
-			} else {
-				$lastID = $dbw->selectRow(
-					'mw_namespaces',
-					'ns_namespace_id',
-					[
-						'ns_dbname' => $wgDBname,
-						'ns_namespace_id >= 3000'
-					],
-					__METHOD__,
-					[
-						'ORDER BY' => 'ns_namespace_id DESC'
-					]
-				);
-
-				if ( $lastID ) {
-					$nsID = [
-						'namespace' => (int)$lastID->ns_namespace_id + 1,
-						'namespacetalk' => (int)$lastID->ns_namespace_id + 2
-					];
-				} else {
-					$nsID = [
-						'namespace' => 3000,
-						'namespacetalk' => 3001
-					];
-				}
-			}
-
 			foreach ( [ 'namespace', 'namespacetalk' ] as $name ) {
-				$nsAlias = (string)json_encode( [] );
-				if ( !empty( $formData["aliases-$name"] ) ) {
-					$nsAlias = (string)json_encode( explode( "\n", $formData["aliases-$name"] ) );
-				}
 				$build[$name] = [
 					'ns_dbname' => $wgDBname,
 					'ns_namespace_id' => $nsID[$name],
 					'ns_namespace_name' => str_replace( ' ', '_', $formData["namespace-$name"] ),
 					'ns_searchable' => (int)$formData["search-$name"],
 					'ns_subpages' => (int)$formData["subpages-$name"],
-					'ns_aliases' => $nsAlias,
+					'ns_aliases' => $formData["aliases-$name"] == "" ? "[]" : json_encode( explode( "\n", $formData["aliases-$name"] ) ),
```

Comment:
Try empty() ? <> : <>;

Reply:
To what end? This does the same thing, and I know it works.